modules/045-snapshot-controller/templates/snapshot-controller/deployment.yaml

```diff
@@ -0,0 +1,102 @@
+{{- if (.Values.global.enabledModules | has "vertical-pod-autoscaler-crd") }}
+---
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: d8-snapshot-controller
+  namespace: kube-system
+  {{- include "helm_lib_module_labels" (list . (dict "app" "d8-snapshot-controller" "workload-resource-policy.deckhouse.io" "master")) | nindent 2 }}
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: Deployment
+    name: d8-snapshot-controller
+  updatePolicy:
+    updateMode: "Auto"
+{{- end }}
+---
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  annotations:
+    helm.sh/hook: post-upgrade, post-install
+    helm.sh/hook-delete-policy: before-hook-creation
+  name: d8-snapshot-controller
+  namespace: kube-system
+  {{- include "helm_lib_module_labels" (list . (dict "app" "d8-snapshot-controller" )) | nindent 2 }}
+spec:
+  minAvailable: {{ include "helm_lib_is_ha_to_value" (list . 1 0) }}
+  selector:
+    matchLabels:
+      app: d8-snapshot-controller
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: d8-snapshot-controller
+  namespace: kube-system
+  {{- include "helm_lib_module_labels" (list . (dict "app" "d8-snapshot-controller")) | nindent 2 }}
+spec:
+  minReadySeconds: 15
+  replicas: {{ include "helm_lib_is_ha_to_value" (list . 2 1) }}
+  selector:
+    matchLabels:
+      app: d8-snapshot-controller
+  template:
+    metadata:
+      labels:
+        app: d8-snapshot-controller
+    spec:
+      imagePullSecrets:
+        - name: deckhouse-registry
+      {{- include "helm_lib_priority_class" (tuple . "cluster-low") | nindent 6 }}
+      {{- include "helm_lib_node_selector" (tuple . "master") | nindent 6 }}
```

Comment:
Why do we need to schedule it on master nodes?

Reply:
Because it is the same controller as any others.
Eg. if we deploy csi-controller on masters, why is snapshot-controller should be exception?

Reply:
We schedule the CSI controller on master nodes because we need to order disks before the cluster becomes bootstrapped. I doubt that we need to make snapshots of these disks at the same time.

Reply:
Ok, should I remove taint as well?

Reply:
There are should be tolerations and a node selector for system nodes.

Reply:
ok, got it!